### PR TITLE
Valko genre parsing

### PR DESCRIPTION
**Pull request by makovako**
- [X] Parsing genres and storing them in DB
- [X] Parsing discription and sotring in DB

### CODE REVIEW

File: service/src/main/java/cz/muni/fi/pv254/parsing/App.java [hunk 1]
```diff
@@ -179,6 +265,7 @@ private RecommendationDTO parseRecommendation(JSONObject review,GameDTO game) {
             rec.setVotedUp(votedUp);
             rec.setVotesUp(votesUp);
             rec.setWeightedVoteScore(weightedVoteScore);
+            // TODO je to lepsie takto setGame, alebo radsej do hry pridat cely list recommendations ?
             rec.setGame(game);
```
Comment (makovako) [inline service/src/main/java/cz/muni/fi/pv254/parsing/App.java]: je lepsie to uukladat ako
recomendation.setgame(game) (Aktualne to takto robim)

alebo az na konci
game.setrecommendations(list of recommendations)

alebo je to jedno

Comment (makovako) [reply service/src/main/java/cz/muni/fi/pv254/parsing/App.java]: @VargaiD

File: service/src/main/java/cz/muni/fi/pv254/parsing/App.java [hunk 2]
```diff
@@ -196,9 +283,16 @@ public int inteligentParse(long gameID) {
...
+            game.setShortDescription(downloadShortDescritpion(gameID));
             game = gameFacade.add(game);
         }
         game = gameFacade.findBySteamId(game.getSteamId());
+        Set<GenreDTO> genres = parseGenres(game);
+        game.setGenres(genres);
+        // TODO toto tu musi byt, inak ak hra nema ziadne reviews neulozia sa zanre
+        // TODO |
+        // TODO V
+        gameFacade.update(game);
```
Comment (makovako) [inline service/src/main/java/cz/muni/fi/pv254/parsing/App.java]: @VargaiD ak tu neurobim update tak ak hra nema ziadnu recommendation tak sa neulozi vazba genres:games
len ci to tu moze takto ostat

File: service/src/main/java/cz/muni/fi/pv254/parsing/App.java [hunk 3]
```diff
@@ -374,6 +468,8 @@ public String downloadUserName(long userId) {
                 Element body = doc.body();
                 Elements nieco = body.getElementsByAttributeValue("class","actual_persona_name");
                 if (nieco.isEmpty()) {
+                    // TODO tu to niekedy pada ze nie je strana dostupna
+                    // TODO aj tu to treba zopakovat {tries}
                     throw new IllegalArgumentException("User Name not found for id "+Long.toString(userId));
```
Comment (makovako) [inline service/src/main/java/cz/muni/fi/pv254/parsing/App.java]: niekedy to vrati nepaltnu stranku ze je pretazeny alebo nedosutpny tak vybehne tato exception a prestane to spracovavat lebo dostane null name
vyriesim to inokedy

File: service/src/main/java/cz/muni/fi/pv254/parsing/App.java (hunk 2 above)
Comment (VargaiD) [reply service/src/main/java/cz/muni/fi/pv254/parsing/App.java]: Myslim ze ok

File: service/src/main/java/cz/muni/fi/pv254/parsing/App.java (hunk 1 above)
Comment (VargaiD) [reply service/src/main/java/cz/muni/fi/pv254/parsing/App.java]: mne sa viac paci game.setrecommendations(list of recommendations) ale ked funguju obe tak je to jedno

File: service/src/main/java/cz/muni/fi/pv254/parsing/App.java [hunk 4]
```diff
@@ -138,13 +142,135 @@ private StringBuffer getJsonFromUrl(String website) {
...
+
+    /**
+     * Loads games from top100.csv file
+     * @return true if successfull
+     */
+    public boolean loadTop100() {
+        System.out.println(System.getProperty("user.dir"));
+        try {
+            // TODO cant find right path to file
+            String file = "${maven.multiModuleProjectDirectory}/service/src/main/resources/top100csv";
```
Comment (makovako) [inline service/src/main/java/cz/muni/fi/pv254/parsing/App.java]: cant find the right path when running in tomcat

**Pull request merged**